osi_logicallane.proto

```diff
@@ -0,0 +1,502 @@
+        // A world position
+        optional Vector3d world_position = 1;
+
+        // S position on the reference line
+        optional double s_position = 2;
+    }
+}
+
+// Like a LaneBoundary, but with a reference and ST positions.
+//
+// Notes on design decisions:
+// - The LogicalLaneBoundary has ST coordinates, and is thus a separate type
+//   from LaneBoundary.
+//   Advantages of this decision:
+//   - Calculations like getting the lane width at a position are easy, since
+//     one can just get the boundary points of the left and right boundary at
+//     the desired S position (via linear interpolation), and calculate the
+//     width from the two points. Also getting something like the distance to
+//     the lane border is very easy.
+//   - No centerline of the lane is necessary, since this can very easily be
+//     generated from the boundaries.
+//   Disadvantages of this decision:
+//   - Lane boundaries cannot be shared with physical lanes. This results in
+//     more data needed. This can mostly be mitigated by only transmitting the
+//     lane boundaries during initialization (e.g. via the OSMP GroundTruthInit
+//     message).
+// - The LogicalLaneBoundary contains all data directly which an agent model is
+//   likely to need. It does not include information normally only used by
+//   sensor models (e.g. the exact length of the color markings on the road).
+//   This information can be gotten from the physical lane referenced in the
+//   LogicalLane, if needed.
+//
+// TODO To Discuss: It would be entirely possible to add ST coordinates also to
+// LaneBoundary. Then no separate type would be needed. However, it is unclear if
+// the additional overhead is acceptable for sensor models.
+//
+message LogicalLaneBoundary
+{
+    // The ID of the lane boundary.
+    //
+    // \rules
+    // is_globally_unique
+    // \endrules
+    //
+    optional Identifier id = 1;
+
+    // points making up the lane boundary.
+    // The boundary must be defined in the same direction as the reference
+    // line. So S positions should increase along the line. Note that S
+    // positions will not always increase strictly monotonically.
+    // Example:
+    //                |---------\
+    //                |          \
+    //  /---boundary--|           \------------
+    // /
+    // ---------- reference line --------------
+    // At the place where the boundary suddenly increases, two points will have
+    // the same S coordinate.
+    //
+    // If the boundary approximates a curve (e.g. a cubic function in
+    // OpenDRIVE), the points must be chosen in a way that the lateral distance
+    // to the ideal line does not exceed 5cm. As shown in the following image:
+    //
+    // \image html line_approximation_error.svg "Approximation error"
+    // Approximation error green line.
+    //
+    // The Z error (difference in Z height between boundary_line and the "real"
+    // line) must also not exceed 5cm.
```

Comment:
I think in the last meeting we said that we should think about if this is a reasonable value for z. The value we decide probably also makes sense for physical lane boundaries.

Reply:
Agreed. I think it depends on the use-case. If doing visualization for humans (e.g. in a driving simulator), 5cm is already quite large, since e.g. a pedestrian will either appear to float, or his shoes will not be completely visible, if the error is 5cm. So for this use-case, the error should be more in the 1-2cm range.

For agent models deciding where and how they can drive, 5cm error also sounds a bit large: imagine two lanes are connected and in reality they have a height difference of 5cm. Humans would drive really slow if the road height changed by 5cm (imagine road works where the road suddenly is 5cm lower). If this is hidden from OSI, then agents will drive far too fast across such an edge. So here, too, 5cm is too large, but probably even 2cm error is too large for this use-case.

So something like 1-2cm for Z error?

Reply:
1-2cm sounds like a good approximation to me. So e.g. speed bumps could be modelled I assume. @arnedueselder what do you think? Do you think that would be a reasonable value for the surface lines as well?

Reply:
Allowed Z error set to 2cm.